### PR TITLE
9165 documentation for parsers

### PR DESCRIPTION
**Pull request by eeliang**
### Which issues does this fix?
https://github.com/workato/issues/issues/9165

### PR submission checklist
- [x] Optimised every new image
- [x] Checked all content for grammatical errors (using Grammarly, for example)
- [x] Added new files to SUMMARY.md

### CODE REVIEW

File: features/handling-csv-files.md [hunk 1]
```diff
@@ -1,86 +1,69 @@
...
+For more information on working with list, see [here](/features/list-management.md)
 
-In the **Parse CSV** action, provide the file content to the CSV body input field, and specify the header line.
+## Parsing CSV files via external parser
+Certain connectors have triggers/actions that can parse CSV content. The [Box connector](/connectors/box.md) has this capability. You can simply listen to **new CSV files** and parse them via the same trigger.
 
-![Utilities action - Parse CSV](/assets/images/features/files-and-attachments/parse-csv-utilities-step.png)
-*Field mapping for the Parse CSV action*
+![CSV file parsing in Box trigger](/assets/images/features/handling-csv-files/csv-file-parsing-recipe.png)
+*CSV file parsing in Box trigger. See the sample recipe [here](https://www.workato.com/recipes/485023)*
```
Comment (graceliching) [inline features/handling-csv-files.md]: *CSV file parsing with a Box trigger. See the sample recipe [here](https://www.workato.com/recipes/485023).*

File: features/handling-soap.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+![Parse SOAP message](/assets/images/features/handling-soap/parse-soap-message.png)
+*Parse SOAP message*
+
+| Input field     | Description |
+| --------------- | ----------- |
+| Sample document | A sample SOAP/XML that has the same format as the SOAP document to be parsed. |
+| Document        | The input SOAP content to be parsed. |
+
+Datapills are generated according to the **sample document** provided. It includes all SOAP tags and all attributes associated with each tag.
```
Comment (graceliching) [inline features/handling-soap.md]: Datapills are generated according to the **Sample document** provided in the input. It includes all SOAP tags and all attributes associated with each tag.

File: features/handling-soap.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+### Output
+The output of this action is a file data object, encoded in SOAP/XML. You can upload this directly to your target web system.
+
+### Setup message template
+To effectively draft a SOAP/XML message, you first have to setup the SOAP/XML schema in Message templates. You can find out more about Message templates [here](features/message-template.md).
+
+Setup a message template for your use case. Define the schema of the output SOAP document and use `{tags}` to indicate where datapills are being used.
+
+![SOAP message template for employee list](/assets/images/features/handling-soap/SOAP-message-template.png)
```
Comment (graceliching) [inline features/handling-soap.md]: Broken image here

File: features/handling-xml.md [hunk 4]
```diff
@@ -1,57 +1,81 @@
...
 
-This is a sample request body to Intacct Web services. The XML data is stored in a top level tag `<request>`, which has nested tags `<control>` and `<operation>`, which in turn, has respective nested tags. Because of this feature, XML data, when coupled with an adequate query language, is capable of storing and transporting deeply nested data.
+The XML data is stored in a top level tag `<Contact>`, which has nested tags `<Name>`, `<Address>`, and `<Phone>`. These tags further contain nested tags. This feature allows XML to create, store and transport deeply nested data.
+
+## XML parser by Workato
+The easiest way to convert raw XML data into usable datapills is to parse it with the built-in XML parser. **XML Parser by Workato** is a native application that does not require any connection setup.
+
+Select **App** > **XML parser by Workato** to get started.
+
+![XML parser by Workato input fields](/assets/images/features/handling-xml/xml-parser-by-workato.png)
```
Comment (graceliching) [inline features/handling-xml.md]: broken image

File: features/handling-yaml.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+</details>
+
+## YAML parser by Workato
+Convert YAML content into useable datapills with the built-in YAML parser. **YAML parser by Workato** is a native application that does not require any connection setup.
+
+Select **App** > **YAML parser by Workato** to get started.
+
+This parser requires you to download the YAML file on to Workato beforehand. You can do so with a connector's built-in download action (e.g. Dropbox) or with [File by Workato](/handling-files-and-attachments.md).
+
+![Parse YAML content](/assets/images/features/handling-yaml/yaml-parser-by-workato.png)
```
Comment (graceliching) [inline features/handling-yaml.md]: broken image

Comment (eeliang) [reply features/handling-yaml.md]: This is due to capitalization in the root directory. e.g. `YAML-parser-by-workato.png`

File: features/handling-xml.md (hunk 4 above)
Comment (eeliang) [reply features/handling-xml.md]: This is due to capitalization in the root directory. e.g. `XML-parser-by-workato.png`

File: features/handling-soap.md (hunk 3 above)
Comment (eeliang) [reply features/handling-soap.md]: This is due to the capitalization in the relative link

**Pull request merged**